nautobot/core/models/__init__.py

```diff
@@ -34,6 +38,31 @@ def present_in_database(self):
         """
         return not self._state.adding
 
+    @classproperty  # https://github.com/PyCQA/pylint-django/issues/240
+    def _content_type(cls):  # pylint: disable=no-self-argument
+        """
+        Return the ContentType of the object, never cached.
+        """
+        return ContentType.objects.get_for_model(cls)
+
+    @classproperty  # https://github.com/PyCQA/pylint-django/issues/240
+    def _content_type_cached(cls):  # pylint: disable=no-self-argument
+        """
+        Return the ContentType of the object, cached.
+        """
+        cache_key = f"{cls._meta.label_lower}._content_type"
+
+        return cache.get_or_set(cache_key, cls._content_type, get_settings_or_config("CONTENT_TYPE_CACHE_TIMEOUT"))
+
+    @classmethod
+    def _get_content_type(cls, skip_cache=False):
```

Comment:
What's the reasoning behind `_`-prefixing these? They don't seem to be private methods/properties since we're using them outside this class/module - would unprefixed names collide with something, or anything like that?